service/history/reset/resetter.go

```diff
@@ -202,32 +202,32 @@ func (r *workflowResetterImpl) prepareResetWorkflow(
 		return nil, err
 	}
 
-	// TODO add checking of reset until event ID == decision task started ID + 1
 	decision, ok := resetMutableState.GetInFlightDecision()
-	if !ok || decision.StartedID+1 != resetMutableState.GetNextEventID() {
-		return nil, &shared.BadRequestError{
-			Message: fmt.Sprintf("Can only reset workflow to DecisionTaskStarted + 1: %v", baseRebuildLastEventID+1),
+	if ok {
```

Comment:
Not sure this will work. InFlightDecision == HasDecisionStarted. 
So there is also case that decision scheduled but not started yet. So if you simply schedule another decision, I am not sure what will happen

Reply:
double check the code that MutableState allows calling scheduleDecisoin multiple times. So probably this is not an issue. But make sure you test this case, and strongly recommend you to add an integration test.

Reply:
Yes. I am testing with a decision task timed out